### PR TITLE
Rails 5

### PR DESCRIPTION
**Pull request by netshade**
Update the site to use Rails 5.1 and Ruby 2.4. Like a true expert I just ran bundle update everything, ran the tests, saw they worked, and felt confident that this would totally work without problems. 

But seriously presented here for review and discussion.

### CODE REVIEW

**Comment by netshade** [comment]
Deprecation warnings on tests are due to `slugged` gem using `changed?`

**Comment by netshade** [comment]
Addresses #23

**Comment by netshade** [comment]
Deploy tasks:

- [ ] Before deploy, ensure `OMG_SECRET_DEVISE_TOKEN` ENV var is set on production

**Comment by netshade** [comment]
Updated after checking out job admin sections, markdown preview was broken. Tested creating jobs and things seem to work okay tho.

**Comment by netshade** [comment]
Some issues showed up only in prod env after a test deploy, corrected.

File: app/models/job.rb [hunk 1]
```diff
@@ -7,7 +7,6 @@ class Job < ActiveRecord::Base
   belongs_to :user
 
   attr_accessor :publish_now
-  attr_accessible :title, :description, :published_at
 
   after_update :notify_if_published
```
Comment (jqr) [inline app/models/job.rb]: This should probably be `after_commit`

**Pull request merged**